test/server/cards/01_SOR/units/SawGerreraExtremist.spec.ts

```diff
@@ -0,0 +1,49 @@
+describe('Saw Gerrera, Extremist', function () {
+    integration(function (contextRef) {
+        it('should not prompt player if no units are available to return to hand', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['surprise-strike'],
+                    groundArena: ['saw-gerrera#extremist'],
+                },
+                player2: {
+                    hand: ['resupply', 'vanquish', 'battlefield-marine'],
+                    resources: ['smugglers-aid', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst'],
+                    hasInitiative: true,
+                }
+            });
+            const { context } = contextRef;
+
+            // opponent play an event : 2 damage on base
+            context.player2.clickCard(context.resupply);
```

Comment:
Could we replace this with an event that has a sub-ability step like `ifYouDo` or `then` to confirm that it's not somehow triggering on each CardAbilityStep instead of the whole CardAbility